src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -846,6 +855,11 @@ internal async Task SearchPackagesAndRefreshUpdateCountAsync(string searchText,
                 // this will wait for searchResultTask to complete instead of creating a new task
                 await _packageList.LoadItemsAsync(loader, loadingMessage, _uiLogger, searchResultTask, _loadCts.Token);
 
+                if (_isTransitiveDependenciesExperimentEnabled && ActiveFilter == ItemFilter.Installed)
+                {
+                    _packageList.AddPackageLevelTypeGrouping();
```

Comment:
I think some of the UI state logic can be moved to XAML.

One way to accomplish that would be to create a Property and then, add a DataTrigger to set Control Template or Style to with desired UI.

See https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageItemDeprecationLabel.xaml

Reply:
We need more control over when the grouping gets cleared and added than that would offer.